### PR TITLE
Distributed iters dynamic initial

### PR DESCRIPTION
**Pull request by s-geronimoanderson**
Distributed dynamic iterator ready for code review. Implementation is 95% similar to distributed guided; Uses new `dynamicSubrange` instead of `guidedSubrange` for subrange calculation.

Includes correctness test and valid input tests (positive and negative). Performance test updates will be in a separate PR.

### CODE REVIEW

File: test/modules/packages/distributedIters/DistributedIters.chpl [hunk 1]
```diff
@@ -17,59 +17,332 @@
...
+*/
 config const infoDistributedIters:bool = false;
 
+// Distributed Dynamic Iterator.
+/*
+  :arg c: The range (or domain) to iterate over. The range (domain) size must
+    be positive.
+  :type c: `range(?)` or `domain`
+
+  :arg chunkSize: The chunk size to yield to each thread. Must be positive.
```
Comment (benharsh) [inline test/modules/packages/distributedIters/DistributedIters.chpl]: We tend to prefer the term ``task`` over ``thread``.

File: test/modules/packages/distributedIters/DistributedIters.chpl [hunk 2]
```diff
@@ -89,12 +362,12 @@ iter distributedGuided(c,
                        coordinated:bool=false,
                        workerLocales=Locales)
 {
-  assert((isDomain(c) || isRange(c)), ("DistributedIters: Serial guided "
-                                       + "iterator: must use a valid domain "
-                                       + "or range"));
-
+  compilerAssert(isDomain(c) || isRange(c),
```
Comment (benharsh) [inline test/modules/packages/distributedIters/DistributedIters.chpl]: Nice 👍

File: test/modules/packages/distributedIters/DistributedIters.chpl [hunk 3]
```diff
@@ -296,6 +570,39 @@ where tag == iterKind.follower
...
+  dynamic schedule would have produced at that point.
+*/
+private proc dynamicSubrange(c:range(?),
+                             chunkSize:int,
+                             stage:int)
+{
+  const low:int = (c.low + (stage * chunkSize));
+  const potentialHigh:int = (low + (chunkSize - 1));
+  const cHigh = c.high;
+  const high:int = if potentialHigh <= cHigh
```
Comment (benharsh) [inline test/modules/packages/distributedIters/DistributedIters.chpl]: Is this just ``min(potentialHigh, cHigh)`` ?

File: test/modules/packages/distributedIters/DistributedIters.chpl (hunk 1 above)
Comment (s-geronimoanderson) [reply test/modules/packages/distributedIters/DistributedIters.chpl]: Missed that one! :)

File: test/modules/packages/distributedIters/DistributedIters.chpl (hunk 3 above)
Comment (s-geronimoanderson) [reply test/modules/packages/distributedIters/DistributedIters.chpl]: ... yes. Yes, it is!

**Pull request merged**